### ISSUE TITLE
fix race melody

### ISSUE THREAD

**Issue by jonaz** [stampzilla/stampzilla-go#106]
```
==================
WARNING: DATA RACE
Write at 0x00c00029e1f8 by main goroutine:
  github.com/jonaz/mdns.(*Server).Shutdown()
      /home/jonaz/go/src/github.com/jonaz/mdns/server.go:91 +0xa7
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain.(*Main).Run()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain/servermain.go:65 +0x3e1
  main.main()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/main.go:19 +0x116

Previous read at 0x00c00029e1f8 by goroutine 21:
  github.com/jonaz/mdns.(*Server).recv()
      /home/jonaz/go/src/github.com/jonaz/mdns/server.go:109 +0xac

Goroutine 21 (running) created at:
  github.com/jonaz/mdns.NewServer()
      /home/jonaz/go/src/github.com/jonaz/mdns/server.go:73 +0x30e
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain.(*Main).Run()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain/servermain.go:51 +0x5a7
  main.main()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/main.go:19 +0x116
==================
Found 2 data race(s)

```

```
==================
WARNING: DATA RACE
Write at 0x00c000284a80 by goroutine 24:
  runtime.mapassign_faststr()
      /usr/lib/go/src/runtime/map_faststr.go:190 +0x0
  github.com/olahol/melody.(*Session).Set()
      /home/jonaz/go/src/github.com/olahol/melody/session.go:194 +0x97
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/handlers.(*secureWebsocketHandler).Message()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/handlers/securewebsockethandler.go:52 +0x187b
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver.(*Webserver).handleMessage.func1()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver/webserver.go:141 +0x227
  github.com/olahol/melody.(*Session).readPump()
      /home/jonaz/go/src/github.com/olahol/melody/session.go:133 +0x454
  github.com/olahol/melody.(*Melody).HandleRequestWithKeys()
      /home/jonaz/go/src/github.com/olahol/melody/melody.go:192 +0x453
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver.(*Webserver).handleWs.func1()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver/webserver.go:233 +0x5e6
  github.com/gin-gonic/gin.(*Context).Next()
      /home/jonaz/go/src/github.com/gin-gonic/gin/context.go:109 +0xc6
  github.com/jonaz/ginlogrus.New.func1()
      /home/jonaz/go/src/github.com/jonaz/ginlogrus/ginlogrus.go:37 +0x726
  github.com/gin-gonic/gin.(*Context).Next()
      /home/jonaz/go/src/github.com/gin-gonic/gin/context.go:109 +0xc6
  github.com/gin-gonic/gin.(*Engine).handleHTTPRequest()
      /home/jonaz/go/src/github.com/gin-gonic/gin/gin.go:383 +0x930
  github.com/gin-gonic/gin.(*Engine).ServeHTTP()
      /home/jonaz/go/src/github.com/gin-gonic/gin/gin.go:349 +0x249
  net/http.serverHandler.ServeHTTP()
      /usr/lib/go/src/net/http/server.go:2741 +0xc4
  net/http.(*conn).serve()
      /usr/lib/go/src/net/http/server.go:1847 +0x80a

Previous read at 0x00c000284a80 by goroutine 9:
  runtime.mapaccess2_faststr()
      /usr/lib/go/src/runtime/map_faststr.go:101 +0x0
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain.broadcastUpdate.func1.1()
      /home/jonaz/go/src/github.com/olahol/melody/session.go:201 +0x1a4
  github.com/olahol/melody.(*hub).run()
      /home/jonaz/go/src/github.com/olahol/melody/hub.go:47 +0xda

Goroutine 24 (running) created at:
  net/http.(*Server).Serve()
      /usr/lib/go/src/net/http/server.go:2851 +0x4c5
  net/http.(*Server).ServeTLS()
      /usr/lib/go/src/net/http/server.go:2891 +0x3a4
  net/http.(*Server).ListenAndServeTLS()
      /usr/lib/go/src/net/http/server.go:3048 +0x15d
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver.(*Webserver).Start.func1()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/webserver/webserver.go:87 +0x145

Goroutine 9 (running) created at:
  github.com/olahol/melody.New()
      /home/jonaz/go/src/github.com/olahol/melody/melody.go:82 +0x36d
  github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain.(*Main).Init()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/servermain/servermain.go:110 +0x267
  main.main()
      /home/jonaz/go/src/github.com/stampzilla/stampzilla-go/nodes/stampzilla-server2/main.go:18 +0x108
==================

```

**Comment by jonaz** [comment]
mdns race fixed by changeing to a maintained fork of mdns ( micro/mdns )

**Comment by stamp** [comment]
Should be fixed when https://github.com/olahol/melody/pull/47 is merged

**Comment by jonaz** [comment]
Possible fix in e4ffcf58022d1aa6b6b674ac156c12336c0cd413
